include/ttmlir/Target/TTNN/program.fbs

```diff
@@ -30,7 +30,11 @@ table ToDeviceOp {
 }
 
 table EmptyOp {
-  device: tt.target.DeviceRef;
+  shape: [int64];
+  dtype: DataType;
+  layout: TensorLayout;
+  device: tt.target.DeviceRef;         // optional
+  memcfg: tt.target.MemoryConfigDesc;  // optional
```

Comment:
Following comments above, these can all be inferred from the `out` tensor ref, otherwise we need to assert that these attributes match exactly the TensorRef attributes.